File: examples/flutter/lib/src/models/tag_list.dart
part of appwrite.models;

/// Tags List
class TagList {
    /// Total number of items available on the server.
    final int sum;
    /// List of tags.
    final List<Tag> tags;

    TagList({
        required this.sum,
        required this.tags,
    });

    factory TagList.fromMap(Map<String, dynamic> map) {
        return TagList(
            sum: map['sum'],
            tags: List<Tag>.from(map['tags'].map((p) => Tag.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "sum": sum,
            "tags": tags.map((p) => p.toMap()),
        };
    }
}

File: examples/swift-server/Sources/AppwriteModels/TagList.swift

/// Tags List
public class TagList {

    /// Total number of items available on the server.
    public let sum: Int

    /// List of tags.
    public let tags: [Tag]

    init(
        sum: Int,
        tags: [Tag]
    ) {
        self.sum = sum
        self.tags = tags
    }

    public static func from(map: [String: Any]) -> TagList {
        return TagList(
            sum: map["sum"] as! Int,
            tags: (map["tags"] as! [[String: Any]]).map { Tag.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "sum": sum as Any,
            "tags": tags.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                    
}